receiver/statsdreceiver/internal/transport/client/client.go

```diff
@@ -51,6 +51,15 @@ func (s *StatsD) connect() error {
 		if err != nil {
 			return err
 		}
+	case "unixgram":
```

Comment:
I see we don't document the `transport` setting at all but I wonder if that's a good chance to do it since we are on it.

Reply:
The `transport` setting comes from the `go.opentelemetry.io/collector/config/confignet` package where it's documented [here](https://pkg.go.dev/go.opentelemetry.io/collector/config/confignet#section-readme)

Reply:
In light of [this comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36608#discussion_r1880020678), I've added some specific documentation for the `statsdreceiver`